alf/networks/encoding_networks.py

```diff
@@ -516,6 +517,15 @@ def __init__(self,
             input_tensor_spec (nested TensorSpec): the (nested) tensor spec of
                 the input. If nested, then ``preprocessing_combiner`` must not be
                 None.
+            output_tensor_spec (None|TensorSpec): spec for the output. If None,
+                the output tensor spec will be assumed as
+                ``TensorSpec((output_size, ))``, where ``output_size`` is
+                inferred from network output. Otherwise, the output tensor
+                spec will be ``output_tensor_spec`` and the network output
+                will be reshaped according to ``output_tensor_spec``.
+                Note that ``output_tensor_spec`` is only used for reshaping
+                the network outputs for interpretation purpose and is not used
+                for specifying any network layers.
```

Comment:
There's some redundancy for setting both `output_tensor_spec` and `last_layer_size`. Do you think reasonable to allow the user not to set last_layer_size when output_tensor_spec is provided?

Reply:
I agree the suggested way could reduce some redundancy.  It might be a good choice in some cases such as the CriticNetwork, but might not be as clear in more general cases, as there will be multiple ways to specify the same network structure, which might introduce some confusions.
Also, there might be cases where the net is specified without last layer but only want to re-interprete the output.

I think still use our original way of specifying the networks, and only resort to output_tensor_spec for interpreting the output when necessary might be less confusing. I might miss something but those are my current thought.